syntax = "proto3";

import "google/protobuf/timestamp.proto";

package keelson.primitives;

message TimestampedBytes {
  // Timestamp when value was valid
  google.protobuf.Timestamp timestamp = 1;
  // The value
  bytes value = 2;
}

message TimestampedDouble {
  // Timestamp when value was valid
  google.protobuf.Timestamp timestamp = 1;
  // The value
  double value = 2;
}


message TimestampedFloat {
  // Timestamp when value was valid
  google.protobuf.Timestamp timestamp = 1;
  // The value
  float value = 2;
}

message TimestampedInt {
  // Timestamp when value was valid
  google.protobuf.Timestamp timestamp = 1;
  // The value
  int32 value = 2;
}

message TimestampedString {
  // Timestamp when value was valid
  google.protobuf.Timestamp timestamp = 1;
  // The value
  string value = 2;
}
